alacritty/src/event.rs

```diff
@@ -74,17 +81,49 @@ impl From<TerminalEvent> for Event {
     }
 }
 
+/// Regex search state.
+pub struct SearchState {
+    /// Search string regex.
+    regex: Option<String>,
+
+    /// Search direction.
+    direction: Direction,
+
+    /// Change in display offset since the beginning of the search.
+    display_offset_delta: isize,
+
+    /// Vi cursor position before search.
+    vi_cursor_point: Point,
+}
+
+impl SearchState {
+    fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Default for SearchState {
+    fn default() -> Self {
+        Self {
+            direction: Direction::Right,
+            display_offset_delta: 0,
+            vi_cursor_point: Point::default(),
+            regex: None,
+        }
+    }
+}
+
 #[derive(Default, Clone, Debug, PartialEq)]
 pub struct DisplayUpdate {
     pub dimensions: Option<PhysicalSize<u32>>,
-    pub message_buffer: bool,
     pub font: Option<Font>,
     pub cursor: bool,
+    pub dirty: bool,
```

Comment:
Could you rename it to something else, since it's really confusing with `terminal.dirty` being used around.

Reply:
How so? I've explicitly named it *because* we already have `terminal.dirty`. The `display_update.dirty` is supposed to act exactly the same way, just for display updates. So as far as I can see it, it's just a common way to track generic demand for updates.

Reply:
I mean it's strange that some of our updates are via `self.cursor() || self.font.is_some()` but the rest is `self.dirty`. So for me, it feels like cursor and font is a subset of dirty, when in reality the goal for dirty was a generic update, which doesn't fit into the rest of groups?

Reply:
Are you suggesting to always update dirty together with the other options?

Reply:
You should either compute dirty based on all values from the display update or update dirty as well and check just for dirty when checking for update and deciding whether we should handle it.

Reply:
Sure, let's always set dirty then.